source/auth/auth.rst

```diff
@@ -683,6 +683,176 @@ mechanism
+
+MONGODB-IAM requires that a client create a randomly generated nonce. It is 
+imperative, for security sake, that this be as secure and truly random as possible. 
+
+All messages between MongoDB clients and servers are sent as base64 encoded BSON V1.1 Objects. 
+All fields in these messages have a "short name" which is used in the serialized 
+BSON representation and a human readable "friendly name" which is used in this document. They are as follows:
+
+==== ==================== ========== ============================================================================================================================================== 
+Name Friendly Name        Type       Description
+==== ==================== ========== ==============================================================================================================================================
+r    client nonce         BinData    32 byte cryptographically secure random number 
+p    gs2-cb-flag          32-bit int The integer representation of the ASCII charater 'n' or 'y', i.e., ``110`` or ``121``
+s    server nonce         BinData    64 bytes total. The first 32 bytes from the client nonce and the second 32 bytes from the server nonce
+h    sts hosts            string     DNS Host name of the STS service 
+a    auth_header          string     Authorization header for `AWS Signature Version 4 <https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html?shortFooter=true>`_
+d    X-AMZ-Date           string     Current date in UTC -see `AWS Signature Version 4 <https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html?shortFooter=true>`_
+t    X-AMZ-Security-Token string     Optional AWS security token
+==== ==================== ========== ============================================================================================================================================== 
+
+Drivers MUST NOT advertise support for channel binding, as the server does
+not support it and legacy servers may fail authentication if drivers advertise
+support. The client-first-message MUST set ``p`` to the integer representation 
+of the ASCII character ``n``, i.e., ``110``.
+
+Conversation
+````````````
+As an example, given an ``r`` value of "enJwWTtNSkR+WztFZCE3d1NWSiMpfU54YCgmPU5lY1Q=", a MONGODB-IAM conversation would appear as
+follows:
+
+| C: ``r=enJwWTtNSkR+WztFZCE3d1NWSiMpfU54YCgmPU5lY1Q=,p=110``
+| S: ``s=enJwWTtNSkR+WztFZCE3d1NWSiMpfU54YCgmPU5lY1RHbnN1IWy6vp7GvmtRmcGWYEtjedGEI0ZXi13r7y4V+A==,h="sts.amazonaws.com"``
+
+.. note::
+	Drivers MUST validate that the server nonce is exactly 64 bytes and the first 32 bytes are the same as the client nonce, i.e., ``r``. Drivers must 
+        also validate that the host is greater than 0 and less than or equal to 255 bytes per `RFC 1035 <https://tools.ietf.org/html/rfc1035>`_. Drivers MUST reject
+        DNS names with empty labels, e.g., "abc..def" and error on any additional fields.
```

Comment:
Did you mean to say FQDNs instead of 'DNS names' here?

Reply:
Done.

Reply:
@prashantmital please comment back or resolve.